File: examples/Node.js/rrclient.js
// Hello World client in Node.js
// Connects REQ socket to tcp://localhost:5559
// Sends "Hello" to server, expects "World" back

var zmq       = require('zmq')
  , requester = zmq.socket('req');

requester.connect('tcp://localhost:5559');
var replyNbr = 0;
requester.on('message', function(msg) {
  console.log('got reply', replyNbr, msg.toString());
  replyNbr += 1;
});

for (var i = 0; i < 10; ++i) {
  requester.send("Hello");
}

File: examples/Free Pascal/rrclient.pas
{
  Hello World client
  Connects REQ socket to tcp://localhost:5559
  Sends "Hello" to server, expects "World" back
  @author cpicanco <cpicanco@ufpa.br>
}
program rrclient;

{$mode objfpc}{$H+}

uses SysUtils, zmq, zmq.helpers, zmq.types;

var
  context, requester: Pointer;
  request_nbr: Integer;
  LString: string;
begin
    context := zmq_ctx_new;

    //  Socket to talk to server
    requester := zmq_socket(context, ZMQ_REQ);
    zmq_connect(requester, 'tcp://localhost:5559');
    for request_nbr := 0 to 9 do
      begin
        s_send(requester, 'Hello');
        LString := s_recv(requester);
        WriteLn(Format('Received reply %d [%s]', [request_nbr, LString]));
        LString := '';
      end;
    zmq_close(requester);
    zmq_ctx_destroy(context);
end.

